### DISCUSSION TITLE
How to use BufferedLogger outside of tests?

### DISCUSSION THREAD

**Discussion by fnqista** [sbt/sbt#6881] [Q&A]
I have a parallel processing task, and when logging text with new lines, it becomes interleaved with other threads' logs. The logger I use is the streams logger.

I have read about the BufferedLogger used in tests - but how can I instantiate it using the streams logger?

**Comment by eed3si9n** [comment]
Log buffering is basically a way of deferring the appender (function that takes actions given log events). `scripted` task also implements log buffering by default it's also a test but it's not `test`.

https://github.com/sbt/sbt/blob/450c87f5769e27e0e1bb42b45bd35fe3986a9359/internal/util-scripted/src/main/scala/sbt/internal/scripted/ScriptedTests.scala#L135-L140

Note that log buffering as it's implemented above takes over the appender behavior, so it should be applied only to your own tasks; otherwise you could end up hijacking appender used for relaying BSP messages etc.

**Reply by fnqista** [reply]
Thanks. I will try it out.

I need it for sbt-assembly, where I do the merge in parallel but have to switch to sequential just to log the entries properly when in debug log mode, which is a little annoying.
